selectivecar.go

```diff
@@ -149,7 +149,7 @@ func (sc SelectiveCarPrepared) Dump(w io.Writer) error {
 		return fmt.Errorf("failed to write car header: %s", err)
 	}
 	for _, c := range sc.cids {
-		blk, err := sc.store.Get(c)
+		blk, err := sc.store.Get(context.TODO(), c)
```

Comment:
is this on purpose? if we're breaking APIs, one could make an argument for adding a ctx parameter to Dump

Reply:
It's a question basically. We're breaking go-datastore which is effecting lots of repos anyway, if you'd like to come along for the ride and add contexts to your interfaces that's 👍. If you'd rather us just use `context.TODO()` in all of the go-datastore calls that's fine too.

Reply:
To clarify: we're intentionally breaking go-ipfs-blockstore and the interfaces you are using here are clones of go-ipfs-blockstore which means we should break those too.

However, whether that means we should also break `Dump()` or the go-car specific functions is your call.

Reply:
I see. We should be phasing out uses of go-car v0 over time, so I don't think it's worth breaking its APIs just to add contexts, but I'm happy to review PRs in the future if it's an immediate need for anyone.

Reply:
> We should be phasing out uses of go-car v0 over time

Do you mean the API, or the format, or both?

The format's not going away, we're always (for the foreseeable future) going to want to have the ability to read and create v0 format files, does the v2 API let us do that? My understanding was that the v2 API ties you in to writing v2 format files, even if it can read CARv1 files. So I've continued to lean on the v0 API to do everything CARv1 related, is that wrong?